servers/rendering/renderer_rd/forward_clustered/render_forward_clustered.cpp

```diff
@@ -2644,6 +2644,31 @@ void RenderForwardClustered::_geometry_instance_add_surface(GeometryInstanceForw
 		}
 		_geometry_instance_add_surface_with_material(ginstance, p_surface, material, next_pass.get_local_index(), storage->material_get_shader_id(next_pass), p_mesh);
 	}
+
```

Comment:
I would probably split this function in two, so you can call it again if a material overlay exists, otherwise you end up having a lot of duplicate code.

Reply:
The only part that can be duplicated is the ``next_pass`` ``while`` loop. But something like ``_geometry_instance_add_next_pass_surfaces_with_material`` would help to avoid code duplication